### DISCUSSION TITLE
Notes! Notes Everywhere! (even for things that do not exist)

### DISCUSSION THREAD

**Discussion by DuaneAbrames** [advplyr/audiobookshelf#2815] [Ideas]
I'd like to see a notes/journal feature.  I want to be able to make personal (private to a user) or public (shared for all users of my server) notes about everything in the database.  I want to be able to write a little review about a book, make a note about a narrator, put in an anecdote about the one time I met that author, write something about a series too.  I should be able to add an unlimited number of notes 'attached' to any object.  When viewing notes, all the appropriate notes should be displayed at whatever level I am at.  If I am looking at a book that is in a series, then I could potentially see notes about that book, the series as a whole, the author(s) and the narrator(s).  I have toyed with the idea of keeping a reading journal for years, and now that I am using ABS to listen to everything, I will have a record from here forward of WHEN I listened, but I find I would also like to be able to write notes to my future self about what I just read, or maybe an alternate reading order for a series, or as I said above, even an anecdote related to something in my server.  

One other thing that might be useful is the ability to add items to the database before the actual files show up.  For instance, I could add a book I don't own yet, or which has not been released yet.  I am in a number of SF/Fantasy book groups, and get a flood of recommendations and other people's impressions of books.  I'd like to be able to add "Want to buy" items to my database.  Maybe a special flag that keeps them from becoming "problem books" with a red (!), and a way to filter the library view to just "want to buy" books.

**Comment by nichwall** [comment]
If you can provide UI mockups for the notes and how they would appear in a web browser or mobile for all of these different views, that would be helpful.

You may want to look at Jelu since that is geared more towards a self hosted Goodreads instead of a media server like ABS.

https://github.com/bayang/jelu

**Comment by ZLoth** [comment]
Feature requests submitted:

Server: https://github.com/advplyr/audiobookshelf/issues/3088
App: https://github.com/advplyr/audiobookshelf-app/issues/1244

**Reply by fivestones** [reply]
I like your requests, essentually for bookmarks with the ability to add a note. I think this may be just one part of or a separate feature from what @DuaneAbrames has requested in this thread, which would include the ability to make notes or comments on books as a whole (rather than attached to a specific timeline position in the case of your requests), as well as making comments on other parts of the database (e.g., authors, series, etc.). I think both types of comments/notes would be great to have and would complement each other.

The other thing that @DuaneAbrames idea has is a social aspect, since if these notes are visible to others they could end up being a way to have a not-real-time conversation about a book or an author or whatnot. You could have a book club by having each person add comments to a book and this would be a discussion connected to that book in particular.

@ZLoth I'm not sure but probably your suggestions could work well with a social component as well, e.g., a setting to make it so that by default anyone else on the server can read your comments and see the bookmarks you've left, and can add their own comments to yours.

**Comment by fivestones** [comment]
I'm also pretty interested in the ability to easily add books for which the book files don't (yet) exist. This would be beautiful for want-to-buy lists or for keeping track of recommendations from others, as @DuaneAbrames has mentioned. I think this might be a second feature request though.
